lib/analyzer/package-managers/apt.ts

```diff
@@ -43,25 +44,53 @@ export function analyzeDistroless(
 
 function parseDpkgFile(text: string): AnalyzedPackageWithVersion[] {
   const pkgs: AnalyzedPackageWithVersion[] = [];
-  let curPkg: any = null;
+  let curPkg: AnalyzedPackageWithVersion | null = null;
   for (const line of text.split("\n")) {
-    curPkg = parseDpkgLine(line, curPkg, pkgs);
+    curPkg = parseDpkgLine(line, curPkg!, pkgs);
+    assignPurl(curPkg);
   }
   return pkgs;
 }
 
+function assignPurl(curPkg: AnalyzedPackageWithVersion): void {
+  if (!curPkg || !curPkg.Name || !curPkg.Version) {
+    return;
+  }
+
+  const pkgName = curPkg.Source
+    ? `${curPkg.Source}/${curPkg.Name}`
+    : curPkg.Name;
+  const qualifiers =
+    curPkg.Source && curPkg.SourceVersion
+      ? {
+          upstream: encodeURIComponent(
```

Comment:
A comment on why we need to encode this would be helpful, I don't think it's obvious :)